### DISCUSSION TITLE
Can you change minio api for from 9000 to other port?

### DISCUSSION THREAD

**Discussion by clarkstyx** [minio/minio#15760] [Q&A] [answered]
I have been trying to create multiple minio docker container with different console ports but cant change the api port from 9000 to other ports. The reason why im doing this is because im currently on a bridge network in docker in which all the minio containters share the same ip with the host. I want to implement replication but i just realized that it only works on port 9000. So if theres another way to change the api port it will definitely solve my problem.

**Comment by harshavardhana** [comment, marked as answer]
--address for API port
--console-address for UI port

**Reply by clarkstyx** [reply]
i did that but cant access the minio browser, also I cannot modify alias using mc console on port 9004
C:\Users\clark>C:\Users\clark\Downloads\mc.exe alias set minio4 http://192.168.0.6:9004 minioadmin minioadmin
mc.exe: <ERROR> Unable to initialize new alias from the provided credentials. Get "http://192.168.0.6:9004/probe-bucket-sign-enpvvm6ecpnb/?location=": dial tcp 192.168.0.6:9004: connectex: No connection could be made because the target machine actively refused it.

**Reply by harshavardhana** [reply]
you need to make sure server is running at port 9004 and also this port in unblocked from firewall.

**Reply by clarkstyx** [reply]
so i tried a different machine, applied the certificates and im getthing this error
C:\Users\clark\Downloads>mc.exe alias set minio1 https://10.10.10.136:9090 minioadmin 1234qwerasdf
mc.exe: <ERROR> Unable to initialize new alias from the provided credentials. Get "https://10.10.10.136:9090": x509: certificate signed by unknown authority.

**Reply by harshavardhana** [reply]
trust the certs put the `public.crt` in  ~/.mc/certs/CAs` folder or use --insecure

**Reply by clarkstyx** [reply]
> trust the certs put the `public.crt` in ~/.mc/certs/CAs` folder or use --insecure

yes it worked using --insecure, thanks!

**Reply by jankap** [reply]
> --address for API port --console-address for UI port

That seems not to work in a docker-compose setting:

![image](https://user-images.githubusercontent.com/12893844/227571768-d8997c5f-6574-41f7-a148-41312f8f01ce.png)

![image](https://user-images.githubusercontent.com/12893844/227571623-25d8258d-b517-476f-a623-b7b31f1c27fa.png)

Where does the port 9000 come from?
